// @regulaforensics/facesdk-webclient/dist/cjs-es5/models/face-sdkresult-code.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Regula FaceSDK Web API
 * Regula FaceSDK Web API
 *
 * The version of the OpenAPI document: 3.2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FaceSDKResultCode = void 0;
/**
 * Result codes from core lib.
 * @export
 * @enum {string}
 */
var FaceSDKResultCode;
(function (FaceSDKResultCode) {
    FaceSDKResultCode[FaceSDKResultCode["FACER_OK"] = 0] = "FACER_OK";
    FaceSDKResultCode[FaceSDKResultCode["FACER_NO_LICENSE"] = 200] = "FACER_NO_LICENSE";
    FaceSDKResultCode[FaceSDKResultCode["FACER_IS_NOT_INITIALIZED"] = 201] = "FACER_IS_NOT_INITIALIZED";
    FaceSDKResultCode[FaceSDKResultCode["FACER_COMMAND_IS_NOT_SUPPORTED"] = 202] = "FACER_COMMAND_IS_NOT_SUPPORTED";
    FaceSDKResultCode[FaceSDKResultCode["FACER_COMMAND_PARAMS_READ_ERROR"] = 203] = "FACER_COMMAND_PARAMS_READ_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_INDEX_FILE_NOT_EXIST"] = 204] = "FACER_SEARCHER_INDEX_FILE_NOT_EXIST";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_INDEX_FILE_READ_ERROR"] = 205] = "FACER_SEARCHER_INDEX_FILE_READ_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_INDEX_FILE_DECRYPTION_ERROR"] = 206] = "FACER_SEARCHER_INDEX_FILE_DECRYPTION_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_INDEX_LOAD_ERROR"] = 207] = "FACER_SEARCHER_INDEX_LOAD_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_WRONG_DESCRIPTOR_VERSION"] = 208] = "FACER_SEARCHER_WRONG_DESCRIPTOR_VERSION";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_INDEX_FILE_ENCRYPTION_ERROR"] = 209] = "FACER_SEARCHER_INDEX_FILE_ENCRYPTION_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_INDEX_FILE_SAVE_ERROR"] = 210] = "FACER_SEARCHER_INDEX_FILE_SAVE_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_IMAGE_DECODE_ERROR"] = 211] = "FACER_SEARCHER_IMAGE_DECODE_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_ID_EXISTS_IN_INDEX"] = 212] = "FACER_SEARCHER_ID_EXISTS_IN_INDEX";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_ID_NOT_IN_INDEX"] = 213] = "FACER_SEARCHER_ID_NOT_IN_INDEX";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_NO_FACES"] = 214] = "FACER_SEARCHER_NO_FACES";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_DESCRIPTOR_EXTRACTOR_ERROR"] = 215] = "FACER_SEARCHER_DESCRIPTOR_EXTRACTOR_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_METADATA_READ_ERROR"] = 216] = "FACER_SEARCHER_METADATA_READ_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_NO_SEARCH_LICENSE"] = 217] = "FACER_SEARCHER_NO_SEARCH_LICENSE";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_DESCRIPTOR_DECODE_ERROR"] = 218] = "FACER_SEARCHER_DESCRIPTOR_DECODE_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_DESCRIPTOR_WRONG_SIZE"] = 219] = "FACER_SEARCHER_DESCRIPTOR_WRONG_SIZE";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_ZIP_IMAGE_READ_ERROR"] = 220] = "FACER_SEARCHER_ZIP_IMAGE_READ_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_ZIP_META_READ_ERROR"] = 221] = "FACER_SEARCHER_ZIP_META_READ_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_SIMILAR_PHOTOS_EXIST"] = 222] = "FACER_SEARCHER_SIMILAR_PHOTOS_EXIST";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_PERSON_ZIP_READ_ERROR"] = 223] = "FACER_SEARCHER_PERSON_ZIP_READ_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_LESS_THAN_TWO_IMAGES_IN_REQUEST"] = 224] = "FACER_LESS_THAN_TWO_IMAGES_IN_REQUEST";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_WRONG_ID"] = 225] = "FACER_SEARCHER_WRONG_ID";
    FaceSDKResultCode[FaceSDKResultCode["FACER_SEARCHER_ID_EXISTS_IN_DELETED"] = 226] = "FACER_SEARCHER_ID_EXISTS_IN_DELETED";
    FaceSDKResultCode[FaceSDKResultCode["FACER_VIDEO_DECODE_ERROR"] = 227] = "FACER_VIDEO_DECODE_ERROR";
    FaceSDKResultCode[FaceSDKResultCode["FACER_NOT_ENOUGH_FRAMES"] = 228] = "FACER_NOT_ENOUGH_FRAMES";
    FaceSDKResultCode[FaceSDKResultCode["FACER_OUTPUT_IS_NOT_DEFINED"] = 229] = "FACER_OUTPUT_IS_NOT_DEFINED";
})(FaceSDKResultCode = exports.FaceSDKResultCode || (exports.FaceSDKResultCode = {}));
