// @regulaforensics/facesdk-webclient/dist/cjs-es5/api/matching-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
* Regula FaceSDK Web API
* Regula FaceSDK Web API
*
* The version of the OpenAPI document: 3.2.0
*
*
* NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
* https://openapi-generator.tech
* Do not edit the class manually.
*/
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MatchingApi = exports.MatchingApiFactory = exports.MatchingApiFp = exports.MatchingApiAxiosParamCreator = void 0;
var axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
var base_js_1 = require("../base.js");
/**
* MatchingApi - axios parameter creator
* @export
*/
exports.MatchingApiAxiosParamCreator = function (configuration) {
    var _this = this;
    return {
        /**
        *
            * @summary Detect facial coordinates
            * @param {DetectRequest} detectRequest
        * @param {*} [options] Override http request option.
        * @throws {RequiredError}
        */
        detect: function (detectRequest, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, query, key, key, headersFromBaseOptions, needsSerialization;
                return __generator(this, function (_a) {
                    // verify required parameter 'detectRequest' is not null or undefined
                    if (detectRequest === null || detectRequest === undefined) {
                        throw new base_js_1.RequiredError('detectRequest', 'Required parameter detectRequest was null or undefined when calling detect.');
                    }
                    localVarPath = "/api/detect";
                    localVarUrlObj = new URL(localVarPath, 'https://example.com');
                    if (configuration) {
                        baseOptions = configuration.baseOptions;
                    }
                    localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                    localVarHeaderParameter = {};
                    localVarQueryParameter = {};
                    localVarHeaderParameter['Content-Type'] = 'application/json';
                    query = new URLSearchParams(localVarUrlObj.search);
                    for (key in localVarQueryParameter) {
                        query.set(key, localVarQueryParameter[key]);
                    }
                    for (key in options.query) {
                        query.set(key, options.query[key]);
                    }
                    localVarUrlObj.search = (new URLSearchParams(query)).toString();
                    headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                    localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                    needsSerialization = (typeof detectRequest !== "string" && !(detectRequest instanceof Uint8Array)) || localVarRequestOptions.headers['Content-Type'] === 'application/json';
                    localVarRequestOptions.data = needsSerialization ? JSON.stringify(detectRequest !== undefined ? detectRequest : {}) : (detectRequest || "");
                    return [2 /*return*/, {
                            url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                            options: localVarRequestOptions,
                        }];
                });
            });
        },
        /**
        *
            * @summary Compare provided face images in all combinations and return similarity score for each pair.
            * @param {MatchRequest} matchRequest
        * @param {*} [options] Override http request option.
        * @throws {RequiredError}
        */
        match: function (matchRequest, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, query, key, key, headersFromBaseOptions, needsSerialization;
                return __generator(this, function (_a) {
                    // verify required parameter 'matchRequest' is not null or undefined
                    if (matchRequest === null || matchRequest === undefined) {
                        throw new base_js_1.RequiredError('matchRequest', 'Required parameter matchRequest was null or undefined when calling match.');
                    }
                    localVarPath = "/api/match";
                    localVarUrlObj = new URL(localVarPath, 'https://example.com');
                    if (configuration) {
                        baseOptions = configuration.baseOptions;
                    }
                    localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                    localVarHeaderParameter = {};
                    localVarQueryParameter = {};
                    localVarHeaderParameter['Content-Type'] = 'application/json';
                    query = new URLSearchParams(localVarUrlObj.search);
                    for (key in localVarQueryParameter) {
                        query.set(key, localVarQueryParameter[key]);
                    }
                    for (key in options.query) {
                        query.set(key, options.query[key]);
                    }
                    localVarUrlObj.search = (new URLSearchParams(query)).toString();
                    headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                    localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                    needsSerialization = (typeof matchRequest !== "string" && !(matchRequest instanceof Uint8Array)) || localVarRequestOptions.headers['Content-Type'] === 'application/json';
                    localVarRequestOptions.data = needsSerialization ? JSON.stringify(matchRequest !== undefined ? matchRequest : {}) : (matchRequest || "");
                    return [2 /*return*/, {
                            url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                            options: localVarRequestOptions,
                        }];
                });
            });
        },
    };
};
/**
* MatchingApi - functional programming interface
* @export
*/
exports.MatchingApiFp = function (configuration) {
    return {
        /**
        *
            * @summary Detect facial coordinates
            * @param {DetectRequest} detectRequest
        * @param {*} [options] Override http request option.
        * @throws {RequiredError}
        */
        detect: function (detectRequest, options) {
            return __awaiter(this, void 0, void 0, function () {
                var localVarAxiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.MatchingApiAxiosParamCreator(configuration).detect(detectRequest, options)];
                        case 1:
                            localVarAxiosArgs = _a.sent();
                            return [2 /*return*/, function (axios, basePath) {
                                    if (axios === void 0) { axios = axios_1.default; }
                                    if (basePath === void 0) { basePath = base_js_1.BASE_PATH; }
                                    var axiosRequestArgs = __assign(__assign({}, localVarAxiosArgs.options), { url: basePath + localVarAxiosArgs.url });
                                    return axios.request(axiosRequestArgs);
                                }];
                    }
                });
            });
        },
        /**
        *
            * @summary Compare provided face images in all combinations and return similarity score for each pair.
            * @param {MatchRequest} matchRequest
        * @param {*} [options] Override http request option.
        * @throws {RequiredError}
        */
        match: function (matchRequest, options) {
            return __awaiter(this, void 0, void 0, function () {
                var localVarAxiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.MatchingApiAxiosParamCreator(configuration).match(matchRequest, options)];
                        case 1:
                            localVarAxiosArgs = _a.sent();
                            return [2 /*return*/, function (axios, basePath) {
                                    if (axios === void 0) { axios = axios_1.default; }
                                    if (basePath === void 0) { basePath = base_js_1.BASE_PATH; }
                                    var axiosRequestArgs = __assign(__assign({}, localVarAxiosArgs.options), { url: basePath + localVarAxiosArgs.url });
                                    return axios.request(axiosRequestArgs);
                                }];
                    }
                });
            });
        },
    };
};
/**
* MatchingApi - factory interface
* @export
*/
exports.MatchingApiFactory = function (configuration, basePath, axios) {
    return {
        /**
        *
            * @summary Detect facial coordinates
            * @param {DetectRequest} detectRequest
        * @param {*} [options] Override http request option.
        * @throws {RequiredError}
        */
        detect: function (detectRequest, options) {
            return exports.MatchingApiFp(configuration).detect(detectRequest, options).then(function (request) { return request(axios, basePath); });
        },
        /**
        *
            * @summary Compare provided face images in all combinations and return similarity score for each pair.
            * @param {MatchRequest} matchRequest
        * @param {*} [options] Override http request option.
        * @throws {RequiredError}
        */
        match: function (matchRequest, options) {
            return exports.MatchingApiFp(configuration).match(matchRequest, options).then(function (request) { return request(axios, basePath); });
        },
    };
};
/**
* MatchingApi - object-oriented interface
* @export
* @class MatchingApi
* @extends {BaseAPI}
*/
var MatchingApi = /** @class */ (function (_super) {
    __extends(MatchingApi, _super);
    function MatchingApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
    *
        * @summary Detect facial coordinates
            * @param {DetectRequest} detectRequest
    * @param {*} [options] Override http request option.
    * @throws {RequiredError}
    * @memberof MatchingApi
    */
    MatchingApi.prototype.detectGen = function (detectRequest, options) {
        var _this = this;
        return exports.MatchingApiFp(this.configuration).detect(detectRequest, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
    *
        * @summary Compare provided face images in all combinations and return similarity score for each pair.
            * @param {MatchRequest} matchRequest
    * @param {*} [options] Override http request option.
    * @throws {RequiredError}
    * @memberof MatchingApi
    */
    MatchingApi.prototype.matchGen = function (matchRequest, options) {
        var _this = this;
        return exports.MatchingApiFp(this.configuration).match(matchRequest, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    return MatchingApi;
}(base_js_1.BaseAPI));
exports.MatchingApi = MatchingApi;
